src/services/fileStorage/SecurityCheckService.js

```diff
@@ -0,0 +1,42 @@
+const { FileModel, SecurityCheckStatusTypes } = require('./model');
+const hooks = require('./hooks/SecurityCheckService.hooks.js');
+const logger = require('../../logger');
+
+const SERVICE_PATH = '/fileStorage/securityCheck/';
+
+class SecurityCheckService {
+	async update(id, data) {
+		if (data.error) {
+			logger.error(`Anti-virus service cannot check file with id ${id}. Error message: '${data.error}'`, data);
+			return Promise.resolve();
+		}
+		const status = data.virus_detected === true
+			? SecurityCheckStatusTypes.BLOCKED
+			: SecurityCheckStatusTypes.VERIFIED;
+		await FileModel.updateOne(
+			{ 'securityCheck.requestToken': id },
+			{
+				$set: {
+					'securityCheck.requestToken': null,
+					'securityCheck.status':	status,
+					'securityCheck.reason': data.virus_signature || 'Clean',
+					'securityCheck.updatedAt': Date.now(),
+				},
+			},
+		).exec();
+		return Promise.resolve();
```

Comment:
return updateOne instead of promise resolve?

Reply:
Mongoose queries are not real promises. Also, I don't want any info about the update to leak out.